File: src/main/java/ru/javawebinar/topjava/model/MealTo.java
package ru.javawebinar.topjava.model;

import java.time.LocalDateTime;
import java.util.function.Supplier;

public class MealTo {
    private final LocalDateTime dateTime;

    private final String description;

    private final int calories;

    private final Supplier<Boolean> exceed;

    public MealTo(LocalDateTime dateTime, String description, int calories, Supplier<Boolean> exceed) {
        this.dateTime = dateTime;
        this.description = description;
        this.calories = calories;
        this.exceed = exceed;
    }

    public Boolean getExceed() {
        return exceed.get();
    }

    @Override
    public String toString() {
        return "UserMealWithExceed{" +
                "dateTime=" + dateTime +
                ", description='" + description + '\'' +
                ", calories=" + calories +
                ", exceed=" + getExceed() +
                '}';
    }
}

File: src/main/java/ru/javawebinar/topjava/to/MealTo.kt
package ru.javawebinar.topjava.to

import ru.javawebinar.topjava.model.Meal
import java.time.LocalDateTime

data class MealTo(
        val id: Int,
        val dateTime: LocalDateTime,
        val description: String,
        val calories: Int,
        val excess: Boolean? = null
) {
    companion object {
        fun from(meal: Meal) = MealTo(
                id = meal.id,
                dateTime = meal.dateTime,
                description = meal.description,
                calories = meal.calories
        )
    }

}